classes/local/apibridge.php

```diff
@@ -1614,6 +1616,21 @@ private function get_acl_rules_for_status($courseid, $visibility, $permanent, $g
                 }
                 break;
             case block_opencast_renderer::HIDDEN:
+                if ($permanent && $forceonhidden) {
```

Comment:
Sorry, i do not have enough knowledge about ACLs yet; are there no non-permenent hidden ACLs, or why do you require `$permanent == true`?

Reply:
Yes, there are non-permanent ACLs (which refer to those that will be used for Students).
And permanent ACLs (refer to those that will be used for Lecturers or Admins)

Permanent ACLs must alway be present in the list no matter the visibility of the event.
However,
NON-Permanent ACLs get removed from the ACL list when the event is hidden.

So here in case of HIDDEN, we only make sure that the permanent one are not empty.

In fact the concept has been implemented to return empty on hidden because in previous stages the permanent ones are already inserted into the ACL list and they won't be changed afterwards. no matter if you toggle the visibility 1000 times.

In this special case, the event is getting duplicated without having the ACLs, therefore the ACL is empty from the beginning! So we need to make that we get permanent ACLs on hidden events as well.

The `$forceonhidden` is only to make sure that we don't accidentally change any behavior of the working code/concept and to use it only for this special case!

Hope that I could explain it clearly!

Reply:
Yes, thank you very much!